src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -0,0 +1,101 @@
+using System.Text;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using NATS.Client;
+
+namespace HealthChecks.Nats
+{
+    /// <summary>
+    /// Health check for Nats Server
+    /// </summary>
+    public sealed class NatsHealthCheck : IHealthCheck, IDisposable
+    {
+        /// <summary>
+        ///  Provides factory methods to create connections to NATS Servers.
+        /// </summary>
+        private static readonly ConnectionFactory _connectionFactory = new ConnectionFactory();
+
+        private readonly NatsOptions _options;
+
+        /// <summary>
+        /// An NATS.Client.IConnection object connected to the NATS server.
+        /// </summary>
+        private IConnection? _connection = null;
+
+        public NatsHealthCheck(NatsOptions natsOptions)
+        {
+            _options = natsOptions;
+        }
+
+        public Task<HealthCheckResult> CheckHealthAsync(
+            HealthCheckContext context,
+            CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                _connection = CreateConnection(_options);
+                if (_connection is null)
+                    throw new ArgumentNullException(nameof(_connection));
+                var healthCheckResult = GetHealthCheckResultFromState(_connection);
+                return Task.FromResult(healthCheckResult);
+            }
+            catch (Exception ex)
+            {
+                var unhealthy = new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+                return Task.FromResult(unhealthy);
+            }
+
+            IConnection CreateConnection(NatsOptions options)
+            {
+                if (options == null)
+                    throw new ArgumentNullException(nameof(options));
+                if (!string.IsNullOrWhiteSpace(options.CredentialsPath))
+                    return _connectionFactory.CreateConnection(options.Url, options.CredentialsPath);
+                if (!string.IsNullOrWhiteSpace(options.Jwt) && !string.IsNullOrWhiteSpace(options.PrivateNKey))
+                    return _connectionFactory.CreateConnection(options.Url, options.Jwt, options.PrivateNKey);
+                return _connectionFactory.CreateConnection(options.Url);
+            }
+
+            HealthCheckResult GetHealthCheckResultFromState(IConnection connection)
```

Comment:
Add explicit `private` accessors to all methods without accessors.

Reply:
Local functions [C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions)

> Unlike a method definition, a local function definition cannot include the member access modifier. \
> Because all local functions are private, including an access modifier, such as the private keyword, \
> generates compiler error CS0106, "The modifier 'private' is not valid for this item."